shared/cluster.go

```diff
@@ -633,3 +633,70 @@ func FetchToken(ip string) (string, error) {
 
 	return token, nil
 }
+
+func CreateSecret(secret, namespace string, print bool) error {
+	kubectl := fmt.Sprintf("kubectl --kubeconfig %s", KubeConfigFile)
+	if namespace == "" {
+		namespace = "default"
+	}
+	cmd := fmt.Sprintf("%s create secret generic %s -n %s --from-literal=mykey=mydata",
+		kubectl, secret, namespace)
+	createStdOut, err := RunCommandHost(cmd)
+	if err != nil {
+		return ReturnLogError("failed to create secret: \n%w", err)
+	}
+	if strings.Contains(createStdOut, "failed to create secret") {
+		return ReturnLogError("failed to create secret: \n%w", err)
+	}
+	if print {
+		LogLevel("DEBUG", "Create Secret Output: %s", createStdOut)
+	}
+
+	return nil
+}
+
+func checkPodStatus(print bool) bool {
+	podsRunningStatus := false
+	pods, errGetPods := GetPods(print)
+	if errGetPods != nil || len(pods) == 0 {
+		LogLevel("DEBUG", "Error getting pods. Retry.")
+		return podsRunningStatus
+	}
+	podReady := 0
+	podNotReady := 0
+	for _, pod := range pods {
+		if pod.Status == "Running" || pod.Status == "Completed" {
+			podReady = podReady + 1
+		} else {
+			podNotReady = podNotReady + 1
+			LogLevel("DEBUG", "Pod Not Ready. Pod details: Name: %s Status: %s", pod.Name, pod.Status)
+		}
+	}
+	if podReady+podNotReady != len(pods) {
+		LogLevel("DEBUG", "Length of pods %d != Ready pods: %d + Not Ready Pods: %d", len(pods), podReady, podNotReady)
+	}
+	if podNotReady == 0 {
+		podsRunningStatus = true
+	}
+
+	return podsRunningStatus
+}
+
+// WaitForPodsRunning Waits for pods to reach running state.
+func WaitForPodsRunning(defaultTime time.Duration, times int) error {
+	var podsRunning bool
+	print := false
```

Comment:
no need of this var 

`podsRunning = checkPodStatus(false)`

Reply:
fixed